pennylane/optimize/adaptive.py

```diff
@@ -31,6 +31,7 @@ def append_gate(tape, params, gates):
         qml.apply(o)
 
     for i, g in enumerate(gates):
+        g = g.__copy__()
```

Comment:
`g = copy.copy(g)` also works here. Not sure if it has any other drawbacks.

Reply:
@soranjh I think `copy` is actually preferred here!